diff --git a/lieer/remote.py b/lieer/remote.py
index 6e3973a..62728f7 100644
--- a/lieer/remote.py
+++ b/lieer/remote.py
@@ -25,12 +25,12 @@ class Remote:
   # * https://stackoverflow.com/questions/19615372/client-secret-in-oauth-2-0?rq=1
   #
   OAUTH2_CLIENT_SECRET = {
-        "client_id":"753933720722-ju82fu305lii0v9rdo6mf9hj40l5juv0.apps.googleusercontent.com",
-        "project_id":"capable-pixel-160614",
+        "client_id":"${CLIENT_ID}",
+        "project_id":"${PROJECT_ID}",
         "auth_uri":"https://accounts.google.com/o/oauth2/auth",
         "token_uri":"https://accounts.google.com/o/oauth2/token",
         "auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs",
-        "client_secret":"8oudEG0Tvb7YI2V0ykp2Pzz9",
+        "client_secret":"${CLIENT_SECRET}",
         "redirect_uris":["urn:ietf:wg:oauth:2.0:oob", "http://localhost"]
     }
 
