diff --git a/lieer/remote.py b/lieer/remote.py
index 6e3973a..ade1082 100644
--- a/lieer/remote.py
+++ b/lieer/remote.py
@@ -9,7 +9,7 @@ from oauth2client.file import Storage
 from pathlib import Path
 
 class Remote:
-  SCOPES = 'https://www.googleapis.com/auth/gmail.readonly https://www.googleapis.com/auth/gmail.labels https://www.googleapis.com/auth/gmail.modify'
+  SCOPES = 'https://www.googleapis.com/auth/gmail.readonly https://www.googleapis.com/auth/gmail.labels https://www.googleapis.com/auth/gmail.modify https://www.googleapis.com/auth/gmail.send'
   APPLICATION_NAME   = 'Lieer'
   CLIENT_SECRET_FILE = None
   authorized         = False
